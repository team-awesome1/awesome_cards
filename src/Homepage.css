/* for small mobile devices */
 
@media (min-width: 280px) {
    body {
        background-image: url("wallpaper.jpg");
        background-size: 200%;
    }
    .Homepage {
        font-family: "Luckiest Guy";
        text-align: center;
        display: flex;
        align-items: center;
        min-height: 100vh;
    }
    .Title {
        font-size: 36vw;
        color: rgb(54, 54, 54);
        margin-top: 20%;
        margin-bottom: 2%;
        text-shadow: 10px 10px 4px gray ;
        -webkit-text-stroke-width: 1.2vw;
        -webkit-text-stroke-color: #fba779;
        transform: rotate(-6deg);
        letter-spacing: -4px;
    }

    .Btn {
        display: block;
        font-size: 3.5vh;
        width: 62%;
        min-width: 62%;
        max-height: 10%;
        margin: 1% auto;
        border: none;
        text-align: center;
        font-family: 'Luckiest Guy', cursive;
        box-shadow: 1vh 1vh .75vh rgba(0,0,0,.4);
    }
    .PlayNow{
        background-color: #fecb10;
    }
    .SignIn{
        background-color: #ff7c9c;
    }
    .HowTo{
        background-color: #a9d7d2;
    }
    footer {
        font-family: Arial, Helvetica, sans-serif;
        font-size: 12px; 
        position: absolute;
        bottom: 0;
        left: 0;
        right: 0;
    }
}

/* For large mobile devices and tablets */

@media (min-width: 540px) {
    body {
        background-size: cover;
    }

    .Title {
        font-size: 22vw;
        -webkit-text-stroke-width: .6vw;
        margin-top: -6%;
    }
   /* .Btn{
        display: block;
        margin: auto;
        width: 45%;
        min-width: 45%;
        font-size: 3vh;
    }*/
}

/* For desktops and extra large tablets */

@media (min-width: 1024px) {
    .Title {
        font-size: 18vw;
        -webkit-text-stroke-width: .4vw;
        margin-top: 4%;
    }
}