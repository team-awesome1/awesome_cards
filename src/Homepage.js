import React from 'react';
import { Link } from 'react-router-dom';
import './styles/Homepage.css';

const Homepage = () => {
  return (
    <div className='Homepage'>
      <div className='container'>
        <h1 className='Title'>WAR!</h1>
        <Link to={'/sign-in'}>
          <button className='Btn SignIn'>Sign In</button>
        </Link>
        <Link to={'/play-now'}>
          <button className='Btn PlayNow'>Play Now</button>
        </Link>
        <Link to={'/instructions'}>
          <button className='Btn HowTo'>How to Play</button>
        </Link>
      </div>
      {/* This attribution for wallpaper image should probably move somewhere else? (Could move to about page) */}
      <footer>
        <a href='https://www.freepik.com/free-vector/background-seamless-pattern-vector-with-cute-memphis_15841841.htm#query=cute%20wallpaper&position=15&from_view=search'>
          Background image by rawpixel.com
        </a>{' '}
        on Freepik
      </footer>
    </div>
  );
};

export default Homepage;
