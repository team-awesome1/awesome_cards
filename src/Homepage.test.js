import React from "react";
import { render } from "@testing-library/react";
import Homepage from "./Homepage";

test("Homepage renders without crashing", () => {
    render(<Homepage />);
});

test("When rendered, Homepage matches snapshot", () => {
    const { asFragment } = render(<Homepage />);
    expect(asFragment()).toMatchSnapshot();
});

test("When rendered, Homepage displays title", () => {
    const { getByText } = render(<Homepage />);
    const title = getByText("WAR!");
    expect(title).toBeInTheDocument();
});