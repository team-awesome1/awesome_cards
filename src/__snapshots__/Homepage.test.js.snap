// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When rendered, Homepage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="OptionsGear"
    >
      <figure>
        <img
          alt="gear wheel"
          class="gearIcon"
          src="/gearIcon.png"
        />
        <figcaption>
          <a
            href="https://icons8.com/icon/qpThLj7vV2Sg/settings"
            rel="noreferrer"
            target="_blank"
          >
            Settings
          </a>
           icon by 
          <a
            href="https://icons8.com"
            rel="noreferrer"
            target="_blank"
          >
            Icons8
          </a>
        </figcaption>
      </figure>
    </div>
    <div>
      <h1
        class="Title"
      >
        WAR!
      </h1>
    </div>
    <div
      class="Login Btn"
    >
      <button>
        Sign in/Sign up
      </button>
    </div>
    <div
      class="PlayNow Btn"
    >
      <button>
        Play Now
      </button>
    </div>
    <div
      class="HowTo Btn"
    >
      <button>
        How to Play
      </button>
    </div>
    <footer>
      <a
        href="https://www.freepik.com/free-vector/background-seamless-pattern-vector-with-cute-memphis_15841841.htm#query=cute%20wallpaper&position=15&from_view=search"
      >
        Background image by rawpixel.com
      </a>
       on Freepik
    </footer>
  </div>
</DocumentFragment>
`;
